### PR TITLE
Add case studies from IstioCon 2021

### PR DESCRIPTION
**Pull request by craigbox**
[X] Docs

### CODE REVIEW

File: data/companies.yml [hunk 1]
```diff
@@ -77,18 +77,12 @@ users:
   - name: "Autodesk"
     logo: "/logos/autodesk.png"
     url: "https://www.autodesk.com"
-  - name: "AutoTrader UK"
```
Comment (shamsher31) [inline data/companies.yml:80]: Why is this removed? I see they are an active users of Isito.

**Review by shamsher31 — COMMENTED** [review]
The case studies boxes are getting too large with this PR,  I think the `Learn more` button should be visible with the first rendered page and users should not have to scroll down to click the `Learn More` button.   

**Before**:
case1


**After**:

case2

File: data/companies.yml (hunk 1 above)
Comment (craigbox) [reply data/companies.yml:80]: Because they're a case study, so they don't need to be in the list of "Also used by" logos below.

Comment (shamsher31) [reply data/companies.yml:80]: fair enough.

File: content/en/about/case-studies/salesforce/index.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+skip_toc: true
+skip_byline: true
+skip_pagenav: true
+skip_feedback: true
+doc_type: article
+sidebar_force: sidebar_case_study
+type: case-studies
+---
+
+Istio and Envoy are foundational building blocks of the Salesforce Service Mesh. In [a presentation at IstioCon 2021](https://events.istio.io/istiocon-2021/sessions/salesforce-service-mesh--our-istio-journey/), Pratima Nambiar walks through Salesforce's service mesh journey. She briefly talks about why they chose the service mesh design pattern, how they initially built it using Envoy and an in-house control plane, and their subsequent pivot to Istio. She discusses how they are currently leveraging Istio and their plan to increase adoption of Istio to further enhance our Service Mesh platform.
```
Comment (frankbu) [inline content/en/about/case-studies/salesforce/index.md:20]:
```suggestion
Istio and Envoy are foundational building blocks of the Salesforce Service Mesh. In [a presentation at IstioCon 2021](https://events.istio.io/istiocon-2021/sessions/salesforce-service-mesh--our-istio-journey/), Pratima Nambiar walks through Salesforce's service mesh journey. She briefly talks about why they chose the service mesh design pattern, how they initially built it using Envoy and an in-house control plane, and their subsequent pivot to Istio. She discusses how they are currently leveraging Istio and their plan to increase adoption of Istio to further enhance their Service Mesh platform.
```

**Comment by craigbox** [comment]
(Setting consent for Frank's suggestion.)

**Comment by craigbox** [comment]
@shamsher31: thanks for the feedback; we've gone for "quality of quote" over scroll necessity for the first round, but we can review this later on.

File: content/en/about/case-studies/atlassian/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+skip_byline: true
+skip_pagenav: true
+skip_feedback: true
+doc_type: article
+sidebar_force: sidebar_case_study
+type: case-studies
+weight: 90
+---
+
+Atlassian has been deploying Envoy to the compute nodes of its internal PaaS over the past 2 years to simplify service-to-service communication for internal developers. As of [their presentation at IstioCon 2021](https://events.istio.io/istiocon-2021/sessions/going-dynamic-with-envoy-at-atlassian/), they deploy Envoy with static configuration and they want to take advantage of dynamic features like client-side routing, direct communication, and fault injection. Atlassian decided Istio was the best choice to deliver this over the next year. Nicolas talk through Atlassian’s journey with service-to-service communication, Envoy and the evolution of their home-grown control planes, then walks through the analysis that led to Istio being the best decision for Atlassian’s business moving forward.
```
Comment (ericvn) [inline content/en/about/case-studies/atlassian/index.md:21]: Nicolas talk**s**

Although maybe this should be talked and then need to change walks later to walked.

File: content/en/about/case-studies/t-mobile/index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+skip_feedback: true
+doc_type: article
+sidebar_force: sidebar_case_study
+type: case-studies
+weight: 60
+---
+
+This is a story of struggle, tradeoffs, and triumphs. Istio, as you may know, is a mission-critical piece of software for securing and connecting microservices across platforms. However, it can be daunting to introduce, operationalize, or adopt it successfully.
+
+In [this talk from IstioCon 2021](https://events.istio.io/istiocon-2021/sessions/adopting-istio-across-100-clusters-at-t-mobile/), Joe Searcy digs into T-Mobile’s journey of adopting Istio across 100+ clusters to support microservices for fraud detection, billing, sales and APIs across many teams.
```
Comment (ericvn) [inline content/en/about/case-studies/t-mobile/index.md:23]: Noting the last command before and/or is inconsistent here. This doc seems to use them, and I suspect it's missing after `sales` if that is the case.

**Review by ericvn — APPROVED** [review]
